runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -364,5 +366,17 @@ public State awaitCompletion() throws Throwable {
       }
       return state;
     }
+
+    @Override
+    public State cancel() throws IOException {
+      throw new UnsupportedOperationException("DirectPipelineResult does not support cancel.");
```

Comment:
Not sure this makes sense in the current instantiation -- this cannot be called until after it's done.

I would probably just throw "Unable to cancel a completed job".